### ISSUE TITLE
AWS Lambda Local debug randomly fails in VS Code

### ISSUE THREAD

**Issue by rushikeshsane95** [aws/aws-toolkit-vscode#1676]
**Desktop (please complete the following information):**

-   OS: macOS Big Sur 11.2.3
-   VS Code version: 1.46.1
-   AWS Toolkit extension version: 1.24.0

**Question**

Error while locally debugging lambda function:
Timeout while waiting for command: "sam local invoke"

Here is the VERBOSE level logs of AWS Toolkit:

2021-04-26 18:40:08 [VERBOSE]: SAM debug: config: "Try AppSync"
2021-04-26 18:40:08 [INFO]: Preparing to debug locally: Lambda "./channel/try_appsync/index.handler"
2021-04-26 18:40:08 [INFO]: Building SAM application...
2021-04-26 18:40:08 [VERBOSE]: running: (not started) [/usr/local/bin/sam build --build-dir /tmp/aws-toolkit-vscode/vsctkMVyqal/output --template /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml --base-dir /Users/rushikeshsane/Documents/live-workflow/functions]
2021-04-26 18:40:08 [INFO]: Running command: (not started) [/usr/local/bin/sam build --build-dir /tmp/aws-toolkit-vscode/vsctkMVyqal/output --template /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml --base-dir /Users/rushikeshsane/Documents/live-workflow/functions]
2021-04-26 18:40:08 [INFO]: Running: (not started) [/usr/local/bin/sam build --build-dir /tmp/aws-toolkit-vscode/vsctkMVyqal/output --template /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml --base-dir /Users/rushikeshsane/Documents/live-workflow/functions]
2021-04-26 18:40:08 [VERBOSE]: CloudFormationTemplateRegistry: manager detected a new file: /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml
2021-04-26 18:40:08 [VERBOSE]: CloudFormationTemplateRegistry: manager detected a change to tracked file: /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml
2021-04-26 18:40:08 [VERBOSE]: stderr: Building codeuri: /Users/rushikeshsane/Documents/live-workflow/functions runtime: nodejs12.x metadata: {} functions: ['functions']

2021-04-26 18:40:08 [VERBOSE]: stderr: Running NodejsNpmBuilder:NpmPack

2021-04-26 18:40:09 [VERBOSE]: CloudFormationTemplateRegistry: manager detected a change to tracked file: /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml
2021-04-26 18:40:38 [VERBOSE]: stderr: Running NodejsNpmBuilder:CopyNpmrc

2021-04-26 18:40:38 [VERBOSE]: stderr: Running NodejsNpmBuilder:CopySource

2021-04-26 18:40:53 [VERBOSE]: stderr: Running NodejsNpmBuilder:NpmInstall

2021-04-26 18:40:56 [VERBOSE]: stderr: Running NodejsNpmBuilder:CleanUpNpmrc

2021-04-26 18:40:58 [VERBOSE]: stdout: 
Build Succeeded

2021-04-26 18:40:58 [VERBOSE]: stdout: 
Built Artifacts  : private/tmp/aws-toolkit-vscode/vsctkMVyqal/output
Built Template   : private/tmp/aws-toolkit-vscode/vsctkMVyqal/output/template.yaml

Commands you can use next
=========================
[*] Invoke Function: sam local invoke -t private/tmp/aws-toolkit-vscode/vsctkMVyqal/output/template.yaml
[*] Deploy: sam deploy --guided --template-file private/tmp/aws-toolkit-vscode/vsctkMVyqal/output/template.yaml
    

2021-04-26 18:40:59 [INFO]: Build complete.
2021-04-26 18:40:59 [INFO]: Starting SAM application locally
2021-04-26 18:40:59 [INFO]: AWS.running.command
2021-04-26 18:40:59 [INFO]: Running: (not started) [/usr/local/bin/sam local invoke functions --template /tmp/aws-toolkit-vscode/vsctkMVyqal/output/template.yaml --event /tmp/aws-toolkit-vscode/vsctkMVyqal/event.json --env-vars /tmp/aws-toolkit-vscode/vsctkMVyqal/env-vars.json -d 5858]
2021-04-26 18:40:59 [VERBOSE]: CloudFormationTemplateRegistry: ,anager detected a deleted file: /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml
2021-04-26 18:41:00 [VERBOSE]: SAM: pid 6876: stderr: Invoking ./channel/try_appsync/index.handler (nodejs12.x)

2021-04-26 18:41:00 [VERBOSE]: SAM: pid 6876: stderr: Skip pulling image and use local one: amazon/aws-sam-cli-emulation-image-nodejs12.x:rapid-1.15.0.

2021-04-26 18:41:00 [VERBOSE]: SAM: pid 6876: stderr: Mounting /tmp/aws-toolkit-vscode/vsctkMVyqal/output/functions as /var/task:ro,delegated inside runtime container

2021-04-26 18:41:30 [ERROR]: Timeout while waiting for command: "sam local invoke"
2021-04-26 18:41:30 [ERROR]: Failed to run SAM application locally: Timeout while waiting for command: "sam local invoke"
2021-04-26 18:41:30 [VERBOSE]: SAM: pid 6876: stderr: Execution of function functions was interrupted

2021-04-26 18:41:30 [INFO]: Waiting for SAM application to start...
2021-04-26 18:41:30 [INFO]: Attaching debugger to SAM application...
2021-04-26 18:41:33 [VERBOSE]: SAM: pid 6876: stderr: No response from invoke container for functions

2021-04-26 18:41:33 [VERBOSE]: SAM: command exited (code: 0): PID 6876: [/usr/local/bin/sam local invoke functions --template /tmp/aws-toolkit-vscode/vsctkMVyqal/output/template.yaml --event /tmp/aws-toolkit-vscode/vsctkMVyqal/event.json --env-vars /tmp/aws-toolkit-vscode/vsctkMVyqal/env-vars.json -d 5858]
2021-04-26 18:41:58 [ERROR]: Retry limit reached while trying to attach the debugger.
2021-04-26 18:41:58 [ERROR]: Unable to attach Debugger. Check AWS Toolkit logs. If it took longer than expected to start, you can still attach.
2021-04-26 18:44:07 [INFO]: Successfully sent a telemetry batch of 3

**Comment by justinmk3** [comment]
what `sam --version` ?

> 2021-04-26 18:41:00 [VERBOSE]: SAM: pid 6876: stderr: Mounting /tmp/aws-toolkit-vscode/vsctkMVyqal/output/functions as /var/task:ro,delegated inside runtime container
> 
> 2021-04-26 18:41:30 [ERROR]: Timeout while waiting for command: "sam local invoke"

Toolkit is waiting for `sam local invoke` which seems to hang. Can you see if "debug" log level reveals more info?

**Comment by rushikeshsane95** [comment]
`sam --version` = `SAM CLI, version 1.15.0`

Debug level logs:

2021-04-26 19:17:39 [VERBOSE]: SAM debug: config: "Try AppSync"
2021-04-26 19:17:39 [INFO]: Preparing to debug locally: Lambda "./channel/try_appsync/index.handler"
2021-04-26 19:17:39 [INFO]: Building SAM application...
2021-04-26 19:17:39 [VERBOSE]: running: (not started) [/usr/local/bin/sam build --debug --build-dir /tmp/aws-toolkit-vscode/vsctkbFOo2t/output --template /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml --base-dir /Users/rushikeshsane/Documents/live-workflow/functions]
2021-04-26 19:17:39 [INFO]: Running command: (not started) [/usr/local/bin/sam build --debug --build-dir /tmp/aws-toolkit-vscode/vsctkbFOo2t/output --template /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml --base-dir /Users/rushikeshsane/Documents/live-workflow/functions]
2021-04-26 19:17:39 [INFO]: Running: (not started) [/usr/local/bin/sam build --debug --build-dir /tmp/aws-toolkit-vscode/vsctkbFOo2t/output --template /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml --base-dir /Users/rushikeshsane/Documents/live-workflow/functions]
2021-04-26 19:17:39 [VERBOSE]: CloudFormationTemplateRegistry: manager detected a new file: /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml
2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,502 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,640 | 'build' command is called

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,642 | No Parameters detected in the template

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,663 | 1 resources found in the template

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,664 | Found Serverless function with name='functions' and CodeUri='/Users/rushikeshsane/Documents/live-workflow/functions'
2021-04-26 19:17:39,664 | No Parameters detected in the template

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,682 | Instantiating build definitions

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,682 | No previous build graph found, generating new one
2021-04-26 19:17:39,683 | Unique function build definition found, adding as new (Function Build Definition: BuildDefinition(nodejs12.x, /Users/rushikeshsane/Documents/live-workflow/functions, Zip, , 01f7180e-e7e1-4e23-ba1f-f240ea31398f, {}, []), Function: Function(name='functions', functionname='functions', runtime='nodejs12.x', memory=None, timeout=None, handler='./channel/try_appsync/index.handler', imageuri=None, packagetype='Zip', imageconfig=None, codeuri='/Users/rushikeshsane/Documents/live-workflow/functions', environment={'Variables': OrderedDict()}, rolearn=None, layers=[], events=None, metadata=None, codesign_config_arn=None))

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,683 | Building codeuri: /Users/rushikeshsane/Documents/live-workflow/functions runtime: nodejs12.x metadata: {} functions: ['functions']

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,683 | Building to following folder /private/tmp/aws-toolkit-vscode/vsctkbFOo2t/output/functions

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,684 | Loading workflow module 'aws_lambda_builders.workflows'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,689 | Registering workflow 'PythonPipBuilder' with capability 'Capability(language='python', dependency_manager='pip', application_framework=None)'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,692 | Registering workflow 'NodejsNpmBuilder' with capability 'Capability(language='nodejs', dependency_manager='npm', application_framework=None)'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,694 | Registering workflow 'RubyBundlerBuilder' with capability 'Capability(language='ruby', dependency_manager='bundler', application_framework=None)'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,697 | Registering workflow 'GoDepBuilder' with capability 'Capability(language='go', dependency_manager='dep', application_framework=None)'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,699 | Registering workflow 'GoModulesBuilder' with capability 'Capability(language='go', dependency_manager='modules', application_framework=None)'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,702 | Registering workflow 'JavaGradleWorkflow' with capability 'Capability(language='java', dependency_manager='gradle', application_framework=None)'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,705 | Registering workflow 'JavaMavenWorkflow' with capability 'Capability(language='java', dependency_manager='maven', application_framework=None)'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,707 | Registering workflow 'DotnetCliPackageBuilder' with capability 'Capability(language='dotnet', dependency_manager='cli-package', application_framework=None)'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,710 | Registering workflow 'CustomMakeBuilder' with capability 'Capability(language='provided', dependency_manager=None, application_framework=None)'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,710 | Found workflow 'NodejsNpmBuilder' to support capabilities 'Capability(language='nodejs', dependency_manager='npm', application_framework=None)'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,711 | Running workflow 'NodejsNpmBuilder'

2021-04-26 19:17:39 [VERBOSE]: stderr: 2021-04-26 19:17:39,711 | Running NodejsNpmBuilder:NpmPack
2021-04-26 19:17:39,711 | NODEJS packaging file:/Users/rushikeshsane/Documents/live-workflow/functions to /var/folders/ls/dfg4zzc11cl5t2_52sqs0wgw0000gn/T/tmptrpvtau5
2021-04-26 19:17:39,711 | executing NPM: ['npm', 'pack', '-q', 'file:/Users/rushikeshsane/Documents/live-workflow/functions']

2021-04-26 19:17:39 [VERBOSE]: CloudFormationTemplateRegistry: manager detected a change to tracked file: /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml
2021-04-26 19:17:59 [VERBOSE]: stderr: 2021-04-26 19:17:59,891 | NODEJS packed to functions-0.0.1.tgz

2021-04-26 19:17:59 [VERBOSE]: stderr: 2021-04-26 19:17:59,891 | NODEJS extracting to /var/folders/ls/dfg4zzc11cl5t2_52sqs0wgw0000gn/T/tmptrpvtau5/unpacked

2021-04-26 19:18:07 [VERBOSE]: stderr: 2021-04-26 19:18:07,839 | NodejsNpmBuilder:NpmPack succeeded

2021-04-26 19:18:07 [VERBOSE]: stderr: 2021-04-26 19:18:07,839 | Running NodejsNpmBuilder:CopyNpmrc
2021-04-26 19:18:07,839 | NodejsNpmBuilder:CopyNpmrc succeeded
2021-04-26 19:18:07,839 | Running NodejsNpmBuilder:CopySource

2021-04-26 19:18:23 [VERBOSE]: stderr: 2021-04-26 19:18:23,926 | NodejsNpmBuilder:CopySource succeeded

2021-04-26 19:18:23 [VERBOSE]: stderr: 2021-04-26 19:18:23,926 | Running NodejsNpmBuilder:NpmInstall
2021-04-26 19:18:23,926 | NODEJS installing in: /private/tmp/aws-toolkit-vscode/vsctkbFOo2t/output/functions
2021-04-26 19:18:23,926 | executing NPM: ['npm', 'install', '-q', '--no-audit', '--no-save', '--production', '--unsafe-perm']

2021-04-26 19:18:26 [VERBOSE]: stderr: 2021-04-26 19:18:26,330 | NodejsNpmBuilder:NpmInstall succeeded

2021-04-26 19:18:26 [VERBOSE]: stderr: 2021-04-26 19:18:26,330 | Running NodejsNpmBuilder:CleanUpNpmrc
2021-04-26 19:18:26,330 | NodejsNpmBuilder:CleanUpNpmrc succeeded

2021-04-26 19:18:28 [VERBOSE]: stdout: 
Build Succeeded

2021-04-26 19:18:28 [VERBOSE]: stdout: 
Built Artifacts  : private/tmp/aws-toolkit-vscode/vsctkbFOo2t/output
Built Template   : private/tmp/aws-toolkit-vscode/vsctkbFOo2t/output/template.yaml

Commands you can use next
=========================
[*] Invoke Function: sam local invoke -t private/tmp/aws-toolkit-vscode/vsctkbFOo2t/output/template.yaml
[*] Deploy: sam deploy --guided --template-file private/tmp/aws-toolkit-vscode/vsctkbFOo2t/output/template.yaml
    

2021-04-26 19:18:28 [VERBOSE]: stderr: 2021-04-26 19:18:28,644 | Sending Telemetry: {'metrics': [{'commandRun': {'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam build', 'duration': 49139, 'exitReason': 'success', 'exitCode': 0, 'requestId': '6791a78e-bf72-4899-99a6-76ab6ed883ad', 'installationId': '5efb89de-cafc-4a1c-b2be-5646e1aab537', 'sessionId': '4baf611b-702c-4c66-9429-55a39e41df1d', 'executionEnvironment': 'CLI', 'pyversion': '3.8.7', 'samcliVersion': '1.15.0'}}]}

2021-04-26 19:18:29 [VERBOSE]: stderr: 2021-04-26 19:18:29,389 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)

2021-04-26 19:18:29 [INFO]: Build complete.
2021-04-26 19:18:29 [DEBUG]: localLambdaRunner.invokeLambdaFunction: Try AppSync
2021-04-26 19:18:29 [INFO]: Starting SAM application locally
2021-04-26 19:18:29 [INFO]: AWS.running.command
2021-04-26 19:18:29 [INFO]: Running: (not started) [/usr/local/bin/sam local invoke --debug functions --template /tmp/aws-toolkit-vscode/vsctkbFOo2t/output/template.yaml --event /tmp/aws-toolkit-vscode/vsctkbFOo2t/event.json --env-vars /tmp/aws-toolkit-vscode/vsctkbFOo2t/env-vars.json -d 5858]
2021-04-26 19:18:29 [VERBOSE]: CloudFormationTemplateRegistry: ,anager detected a deleted file: /Users/rushikeshsane/Documents/live-workflow/functions/app___vsctk___template.yaml
2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,052 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics

2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,052 | local invoke command is called

2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,054 | No Parameters detected in the template

2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,080 | 1 resources found in the template

2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,081 | Found Serverless function with name='functions' and CodeUri='functions'

2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,088 | Found one Lambda function with name 'functions'

2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,089 | Invoking ./channel/try_appsync/index.handler (nodejs12.x)
2021-04-26 19:18:30,089 | Environment variables overrides data is standard format
2021-04-26 19:18:30,089 | Loading AWS credentials from session with profile 'None'

2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,101 | Resolving code path. Cwd=/tmp/aws-toolkit-vscode/vsctkbFOo2t/output, CodeUri=functions

2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,101 | Resolved absolute path to code is /tmp/aws-toolkit-vscode/vsctkbFOo2t/output/functions
2021-04-26 19:18:30,101 | Code /tmp/aws-toolkit-vscode/vsctkbFOo2t/output/functions is not a zip/jar file

2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,127 | Skip pulling image and use local one: amazon/aws-sam-cli-emulation-image-nodejs12.x:rapid-1.15.0.


2021-04-26 19:18:30 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:30,127 | Mounting /tmp/aws-toolkit-vscode/vsctkbFOo2t/output/functions as /var/task:ro,delegated inside runtime container

2021-04-26 19:18:32 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:18:32,518 | Setting up SIGTERM interrupt handler

2021-04-26 19:19:02 [ERROR]: Timeout while waiting for command: "sam local invoke"
2021-04-26 19:19:02 [ERROR]: Failed to run SAM application locally: Timeout while waiting for command: "sam local invoke"
2021-04-26 19:19:02 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:19:02,519 | Execution of function functions was interrupted

2021-04-26 19:19:02 [INFO]: Waiting for SAM application to start...
2021-04-26 19:19:02 [DEBUG]: telemetry: passive=0 lambda_invokeLocal
2021-04-26 19:19:02 [DEBUG]: localLambdaRunner.attachDebugger: startDebugging with config: {
  "type": "node",
  "request": "attach",
  "name": "Try AppSync",
  "invokeTarget": {
    "target": "code",
    "lambdaHandler": "./channel/try_appsync/index.handler",
    "projectRoot": "functions"
  },
  "lambda": {
    "runtime": "nodejs12.x",
    "environmentVariables": {},
    "payload": {
      "path": "parser_output.json"
    }
  },
  "__configurationTarget": 5,
  "codeRoot": "/Users/rushikeshsane/Documents/live-workflow/functions",
  "workspaceFolder": {
    "uri": {
      "$mid": 1,
      "fsPath": "/Users/rushikeshsane/Documents/live-workflow",
      "external": "file:///Users/rushikeshsane/Documents/live-workflow",
      "path": "/Users/rushikeshsane/Documents/live-workflow",
      "scheme": "file"
    },
    "name": "live-workflow",
    "index": 0
  },
  "runtime": "nodejs12.x",
  "runtimeFamily": 2,
  "handlerName": "./channel/try_appsync/index.handler",
  "documentUri": {
    "$mid": 1,
    "fsPath": "/Users/rushikeshsane/Documents/live-workflow/functions/channel/try_appsync/index.js",
    "external": "file:///Users/rushikeshsane/Documents/live-workflow/functions/channel/try_appsync/index.js",
    "path": "/Users/rushikeshsane/Documents/live-workflow/functions/channel/try_appsync/index.js",
    "scheme": "file"
  },
  "templatePath": "/tmp/aws-toolkit-vscode/vsctkbFOo2t/output/template.yaml",
  "eventPayloadFile": "/tmp/aws-toolkit-vscode/vsctkbFOo2t/event.json",
  "envFile": "/tmp/aws-toolkit-vscode/vsctkbFOo2t/env-vars.json",
  "debugPort": 5858,
  "useIkpdb": false,
  "baseBuildDir": "/tmp/aws-toolkit-vscode/vsctkbFOo2t",
  "address": "localhost",
  "port": 5858,
  "localRoot": "/Users/rushikeshsane/Documents/live-workflow/functions",
  "remoteRoot": "/var/task",
  "protocol": "inspector",
  "stopOnEntry": false,
  "skipFiles": [
    "/var/runtime/node_modules/**/*.js",
    "<node_internals>/**/*.js"
  ]
}
2021-04-26 19:19:02 [INFO]: Attaching debugger to SAM application...
2021-04-26 19:19:05 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:19:05,431 | Container was not created. Skipping deletion

2021-04-26 19:19:05 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:19:05,431 | Cleaning all decompressed code dirs
2021-04-26 19:19:05,431 | No response from invoke container for functions

2021-04-26 19:19:05 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:19:05,432 | Sending Telemetry: {'metrics': [{'commandRun': {'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam local invoke', 'duration': 35378, 'exitReason': 'success', 'exitCode': 0, 'requestId': '6ddae9e6-eec8-46e4-acbc-f373c697c531', 'installationId': '5efb89de-cafc-4a1c-b2be-5646e1aab537', 'sessionId': 'cba65685-16ac-4961-8a51-dcf5c0d7cfe3', 'executionEnvironment': 'CLI', 'pyversion': '3.8.7', 'samcliVersion': '1.15.0'}}]}

2021-04-26 19:19:06 [VERBOSE]: SAM: pid 7401: stderr: 2021-04-26 19:19:06,076 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)

2021-04-26 19:19:06 [VERBOSE]: SAM: command exited (code: 0): PID 7401: [/usr/local/bin/sam local invoke --debug functions --template /tmp/aws-toolkit-vscode/vsctkbFOo2t/output/template.yaml --event /tmp/aws-toolkit-vscode/vsctkbFOo2t/event.json --env-vars /tmp/aws-toolkit-vscode/vsctkbFOo2t/env-vars.json -d 5858]
2021-04-26 19:19:15 [DEBUG]: attachDebugger: retrying...
2021-04-26 19:19:30 [ERROR]: Retry limit reached while trying to attach the debugger.
2021-04-26 19:19:30 [DEBUG]: telemetry: passive=0 sam_attachDebugger
2021-04-26 19:19:30 [ERROR]: Unable to attach Debugger. Check AWS Toolkit logs. If it took longer than expected to start, you can still attach.
2021-04-26 19:19:36 [DEBUG]: telemetry: passive=0 vscode_viewLogs
2021-04-26 19:20:04 [INFO]: Successfully sent a telemetry batch of 7

**Comment by justinmk3** [comment]
> `sam --version` = `SAM CLI, version 1.15.0`

Latest version is 1.23, try updating `sam`.

If you start a terminal in the same directory as the template.yaml, does `sam local invoke` work?

**Comment by rushikeshsane95** [comment]
I have upgraded sam to latest version now. 

The thing is this error keeps popping up intermittently and it works usually after VS code restart. sam local invoke works in the terminal. I guess it could be because after 1 run of debug, VS code or sam (not sure whom to blame here :-p) might be keeping debugging ports busy for a while. This could give timeout error in the next debug run.

I will wait and try a few times now as I have just upgraded sam version. Hopefully this will resolve the issue. Thanks for the quick turnaround, @justinmk3 :-)
